src/Common/src/Interop/Unix/System.Private.CoreLib.Native/Interop.MemReAlloc.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal unsafe partial class Sys
+    {
+        [DllImport(Interop.Libraries.CoreLibNative, EntryPoint = "CoreLibNative_MemReAlloc")]
+        internal static extern IntPtr MemReAlloc(IntPtr ptr, UIntPtr newSize);
+        
+        [DllImport(Interop.Libraries.CoreLibNative, EntryPoint = "CoreLibNative_MemGetAllocatedSize")]
+        internal static extern IntPtr MemGetAllocatedSize(IntPtr ptr);
```

Comment:
If this is the problem we can easily work around that when we are allocating the original memory (when calling malloc) by zero out the extra allocated memory all the time so when calling realloc we'll be sure we didn't miss any part of the buffer to zero it out. 
I am looking at coreclr and I am not seeing we support realloc with zero memory flag. I looked at the file pal\src\memory\heap.cpp. looks to me in coreclr there were no places calling which need to realloc with zero memory but this is looks not the case in corert which S.P.Interop using this scenario